pkg/ebpf/telemetry/errors_telemetry.go

```diff
@@ -0,0 +1,290 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package telemetry
+
+import (
+	"errors"
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"sync"
+	"syscall"
+	"unsafe"
+
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sys/unix"
+
+	"github.com/DataDog/datadog-agent/pkg/network/ebpf/probes"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+const (
+	maxErrno    = 64
+	maxErrnoStr = "other"
+
+	ebpfMapTelemetryNS    = "ebpf_maps"
+	ebpfHelperTelemetryNS = "ebpf_helpers"
+)
+
+const (
+	readIndx int = iota
+	readUserIndx
+	readKernelIndx
+	skbLoadBytes
+	perfEventOutput
+)
+
+var ebpfMapOpsErrorsGauge = prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfMapTelemetryNS), "Failures of map operations for a specific ebpf map reported per error.", []string{"map_name", "error"}, nil)
+var ebpfHelperErrorsGauge = prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfHelperTelemetryNS), "Failures of bpf helper operations reported per helper per error for each probe.", []string{"helper", "probe_name", "error"}, nil)
+
+var helperNames = map[int]string{
+	readIndx:        "bpf_probe_read",
+	readUserIndx:    "bpf_probe_read_user",
+	readKernelIndx:  "bpf_probe_read_kernel",
+	skbLoadBytes:    "bpf_skb_load_bytes",
+	perfEventOutput: "bpf_perf_event_output",
```

Comment:
not necessary for this PR, but at some point we should consider adding the `bpf_ringbuf_output` as well

Reply:
good point! I'll open a Jira ticket on that request, but yeah it is not in the scope of this PR